source/dbo/Functions/Power BI Reporting/ufnGetCaseRecordChestXrayResults.sql

```diff
@@ -0,0 +1,25 @@
+﻿CREATE FUNCTION [dbo].[ufnGetCaseRecordChestXrayResults](
+	@NotificationId INT,
+	@DiagnosisDate DATETIME2
+)
+RETURNS TABLE
+AS
+RETURN (
+	WITH rankedResults AS
+	(
+	SELECT mr.NotificationId
+	,mr.Result AS result
+	,ROW_NUMBER() OVER
+		(PARTITION BY mr.NotificationId
+		ORDER BY ABS(DATEDIFF(day, mr.TestDate, @DiagnosisDate)) ASC
+		,resultLookup.Ranking ASC
+		) AS rn
+	FROM [$(NTBS)].[dbo].[ManualTestResult] mr
+		LEFT JOIN ChestXrayResultLookup resultLookup ON resultLookup.Result = mr.Result
+	WHERE mr.ManualTestTypeId = 4
+	)
+	SELECT ChestXRayResult = resultLookup.FormattedResult
+	FROM rankedResults
+		LEFT JOIN ChestXrayResultLookup resultLookup ON resultLookup.Result = rankedResults.result
+	WHERE rankedResults.NotificationId = @NotificationId AND rankedResults.rn = 1
```

Comment:
As discussed, you could replace this with a more simple `SELECT TOP(1)` query without the partitioning trick, since you only care about a single notification to begin with. I don't know what affect it would have on the performance, though, but maybe worth a try?

Reply:
Yep I've done that. It's hard to test the performance implications of this change, it seemed ok locally but that doesn't mean very much. Hopefully it should be fine for now and its something we can revisit if we do have performance issues.